elevation_mapping/src/sensor_processors/SensorProcessorBase.cpp

```diff
@@ -92,31 +97,33 @@ bool SensorProcessorBase::process(const PointCloudType::ConstPtr pointCloudInput
 bool SensorProcessorBase::updateTransformations(const ros::Time& timeStamp) {
   const Parameters parameters{parameters_.getData()};
   try {
-    transformListener_.waitForTransform(sensorFrameId_, generalParameters_.mapFrameId_, timeStamp, ros::Duration(1.0));
-
-    tf::StampedTransform transformTf;
-    transformListener_.lookupTransform(generalParameters_.mapFrameId_, sensorFrameId_, timeStamp, transformTf);
-    poseTFToEigen(transformTf, transformationSensorToMap_);
 
-    transformListener_.lookupTransform(generalParameters_.robotBaseFrameId_, sensorFrameId_, timeStamp,
-                                       transformTf);  // TODO(max): Why wrong direction?
-    Eigen::Affine3d transform;
-    poseTFToEigen(transformTf, transform);
-    rotationBaseToSensor_.setMatrix(transform.rotation().matrix());
-    translationBaseToSensorInBaseFrame_.toImplementation() = transform.translation();
-
-    transformListener_.lookupTransform(generalParameters_.mapFrameId_, generalParameters_.robotBaseFrameId_, timeStamp,
-                                       transformTf);  // TODO(max): Why wrong direction?
-    poseTFToEigen(transformTf, transform);
-    rotationMapToBase_.setMatrix(transform.rotation().matrix());
-    translationMapToBaseInMapFrame_.toImplementation() = transform.translation();
+    geometry_msgs::TransformStamped transformGeom;
+    transformGeom = transformBuffer_.lookupTransform(generalParameters_.mapFrameId_, sensorFrameId_, timeStamp, ros::Duration(1.0));
+    transformationSensorToMap_ = tf2::transformToEigen(transformGeom);
+
+    transformGeom = transformBuffer_.lookupTransform(generalParameters_.robotBaseFrameId_, sensorFrameId_, timeStamp,
+                                                      ros::Duration(1.0));  // TODO(max): Why wrong direction?
```

Comment:
@ecthelion99 - What is this TODO? Did you put it here or was it in existing code?

Reply:
These were all existing